src/uptime.c

```diff
@@ -86,46 +86,38 @@ static int uptime_init(void) /* {{{ */
  */
 
 #if KERNEL_LINUX
-  unsigned long starttime;
-  char buffer[1024];
+  double uptime_seconds;
   int ret;
   FILE *fh;
 
   ret = 0;
 
-  fh = fopen(STAT_FILE, "r");
+  fh = fopen(UPTIME_FILE, "r");
 
   if (fh == NULL) {
     char errbuf[1024];
-    ERROR("uptime plugin: Cannot open " STAT_FILE ": %s",
+    ERROR("uptime plugin: Cannot open " UPTIME_FILE ": %s",
           sstrerror(errno, errbuf, sizeof(errbuf)));
     return -1;
   }
 
-  while (fgets(buffer, 1024, fh) != NULL) {
-    /* look for the btime string and read the value */
-    ret = sscanf(buffer, "btime %lu", &starttime);
-    /* avoid further loops if btime has been found and read
-     * correctly (hopefully) */
-    if (ret == 1)
-      break;
-  }
+  ret = fscanf(fh, "%lf", &uptime_seconds);
```

Comment:
Please use the `parse_value_file()` function from `"common.h"` (`src/daemon/common.h`).

Reply:
@octo `uptime_seconds` is an intermediate variable used to calculate the static `boottime = time(NULL) - (long)uptime_seconds;`. Do you still think it's relevant?

Reply:
The original code I replaced was

```c
  while (fgets(buffer, 1024, fh) != NULL) { 
    /* look for the btime string and read the value */ 
    ret = sscanf(buffer, "btime %lu", &starttime); 
    /* avoid further loops if btime has been found and read 
     * correctly (hopefully) */ 
    if (ret == 1) 
      break; 
  } 
```

Reply:
TL;DR: `fscanf()` if fine here, I was confused about the content of `/prod/uptime`.

Previously, the code scanned for a specific line which had the `btime` prefix. Your new code appeared to read a file that contained a *single value* only – which is a common use case and we have the `parse_value_file()` function for it.

Looking at `/proc/uptime` though, it appears that there are two values and the convenience function cannot be used (it complains when there are trailing characters). `fscanf()` is perfectly fine for dealing with this. I briefly pondered whether we should read the second value into an anonymous var just to make it explicit that we're ignoring it, but I think that would make the code harder to read rather than easier.

Reply:
Yep, now I see what you mean. So, do you want me to change something else then? I'm a bit confused whether my implementation is fine or not. To me - it's okay'ish :-)